portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -41,9 +42,21 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
     PreparedQuery results = datastore.prepare(query);
 
+    // Convert the input into an int.
+    String maxCommentsString = request.getParameter("max-comments");
+    int maxComments = 3;  // 3 is the default value.
+    try {
+      maxComments = Integer.parseInt(maxCommentsString);
+    } catch (NumberFormatException e) {
+      System.err.println("Could not convert to int: " + maxCommentsString);
+    }
+
     List<String> comments = new ArrayList<>();
-    for (Entity entity : results.asIterable()) {
-      comments.add((String) entity.getProperty("title"));
+    Iterator<Entity> resultsIterator = results.asIterator();
+    for (int i = 0; i < maxComments; i++) {
+      if (resultsIterator.hasNext()){
+          comments.add((String) resultsIterator.next().getProperty("title"));
+      }   
     }
```

Comment:
I know I said before to avoid 'break' in loops if possible but I think it actually may be warranted in this case since you have an iterator.  I still like this style of for loop better than what you had before because the previous way you had to manually increment a counter.  However, I think for efficiency reasons we should now add back the break (imagine the user asked for 10^9 results but there were only 4 comments, this would cause the page to freeze).  I would probably do the following

`    
for (int i = 0; i < maxComments; i++) {
  if (!resultsIterator.hasNext()){
    break;
  }
  comments.add((String) resultsIterator.next().getProperty("title")); 
}
`

Reply:
Ugh, the comment didn't format properly, but hopefully you can still read it.

Reply:
That's alright, I can still read it.